engine/src/test/java/org/camunda/bpm/engine/test/api/multitenancy/tenantcheck/MultiTenancyUserOperationLogTenantCheckTest.java

```diff
@@ -0,0 +1,352 @@
+  protected RuntimeService runtimeService;
+  protected IdentityService identityService;
+
+  @Before
+  public void init() {
+    taskService = engineRule.getTaskService();
+    historyService = engineRule.getHistoryService();
+    runtimeService = engineRule.getRuntimeService();
+    identityService = engineRule.getIdentityService();
+  }
+
+  @Test
+  public void shouldSetAnnotationWithoutTenant() {
+    // given
+    testRule.deploy(MODEL);
+    identityService.setAuthentication(USER_ONE, null);
+
+    runtimeService.startProcessInstanceByKey(PROCESS_NAME);
+    String taskId = taskService.createTaskQuery().singleResult().getId();
+
+    taskService.complete(taskId);
+
+    UserOperationLogEntry singleResult = historyService.createUserOperationLogQuery().entityType(EntityTypes.TASK).singleResult();
+
+    // when
+    historyService.setAnnotationForOperationLogById(singleResult.getOperationId(), AN_ANNOTATION);
+    singleResult = historyService.createUserOperationLogQuery()
+        .entityType(EntityTypes.TASK)
+        .singleResult();
+
+    // then
+    assertThat(singleResult.getTenantId()).isEqualTo(null);
+  }
+
+  @Test
+  public void shouldSetAnnotationWithTenant() {
+    // given
+    testRule.deployForTenant(TENANT_ONE, MODEL);
+    identityService.setAuthentication(USER_ONE, null, Arrays.asList(TENANT_ONE));
+
+    runtimeService.startProcessInstanceByKey(PROCESS_NAME);
+    String taskId = taskService.createTaskQuery().singleResult().getId();
+
+    taskService.complete(taskId);
+
+    UserOperationLogEntry singleResult = historyService.createUserOperationLogQuery().entityType(EntityTypes.TASK).singleResult();
+
+    // when
+    historyService.setAnnotationForOperationLogById(singleResult.getOperationId(), AN_ANNOTATION);
+    singleResult = historyService.createUserOperationLogQuery()
+        .entityType(EntityTypes.TASK)
+        .singleResult();
+
+    // then
+    assertThat(singleResult.getTenantId()).isEqualTo(TENANT_ONE);
+  }
+
+  @Test
+  public void shouldThrowExceptionWhenSetAnnotationWithDifferentTenant() {
+    // given
+    testRule.deployForTenant(TENANT_ONE, MODEL);
+    identityService.setAuthentication(USER_ONE, null, Arrays.asList(TENANT_ONE));
+
+    runtimeService.startProcessInstanceByKey(PROCESS_NAME);
+    String taskId = taskService.createTaskQuery().singleResult().getId();
+
+    taskService.complete(taskId);
+
+    UserOperationLogEntry singleResult = historyService.createUserOperationLogQuery().entityType(EntityTypes.TASK).singleResult();
+
+    identityService.setAuthentication(USER_TWO, null, Arrays.asList(TENANT_TWO));
+
+    // when/then
+    assertThatThrownBy(() -> historyService.setAnnotationForOperationLogById(singleResult.getOperationId(), AN_ANNOTATION))
+    .isInstanceOf(ProcessEngineException.class)
+    .hasMessageContaining("Cannot update the user operation log entry '"
+        + singleResult.getId() +"' because it belongs to no authenticated tenant.");
```

Comment:
Indentation seems to be off here. This is repeated a couple of times in this class.